### PR TITLE
#148227437 Create Add Skill Form

### PR DESCRIPTION
**Pull request by Mbarak-Mbigo**
#### What does this PR do?
Create add new skill form interface

#### Description of Task to be completed?
Create an interface for adding new skills

#### How should this be manually tested?
After cloning the repo

- Run yarn install
- Run yarn start
- Access the landing page via localhost:3000 and click on the LOGIN TO CONTRIBUTE button.
- Navigate to contribute skill page via New Skill button on the Page menu drop-down
- Add a new skill

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[148227437](https://www.pivotaltracker.com/story/show/148227437)
#### Screenshots (if appropriate)
screen shot 2017-08-16 at 14 58 47
screen shot 2017-08-16 at 14 59 02
screen shot 2017-08-16 at 15 00 04
screen shot 2017-08-16 at 15 00 14

#### Questions:
N/A

### CODE REVIEW

**Comment by andela-cofor** [comment]
11 files is a lot to change, we would have broken this story down, @andela-kakpobome what do you think ?

File: src/components/contributions/contributionNav/ContributionNav.component.tsx [hunk 1]
```diff
@@ -36,21 +37,21 @@ export class ContributionNav extends React.Component <IContributionNavProps, ICo
         <div className="contribution-categories">
           <div className={this.contributionClass('idea')} onClick={() => this.changeActiveForm('idea')}>
             <div className="header">
-              <i className="material-icons">lightbulb_outlines</i>
+              <Icon name="bulb" color="#9b9b9b" />
```
Comment (andela-cofor) [inline src/components/contributions/contributionNav/ContributionNav.component.tsx]: I don't think we can style in a component, let's take this styling to css file

File: src/components/contributions/skillForm/SkillForm.scss [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+        a {
+          position: absolute;
+          top: 0;
+          right: 0;
+
+          &:hover {
+            cursor: pointer;
+            color: map_get($colors, danger);
+            transition: color ease-in 300ms;
```
Comment (andela-cofor) [inline src/components/contributions/skillForm/SkillForm.scss]: color comes before cursor, let's re-arrange

File: src/components/contributions/skillForm/SkillForm.component.tsx [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+    this.handleInput = this.handleInput.bind(this);
+    this.handleTags = this.handleTags.bind(this);
+    this.handleAddOkbb = this.handleAddOkbb.bind(this);
+    this.removeOkbb = this.removeOkbb.bind(this);
+    this.aggregateSkill = this.aggregateSkill.bind(this);
+    this.submitSkill = this.submitSkill.bind(this);
+  }
+
+  public handleInput(value, property): void {
```
Comment (andela-cofor) [inline src/components/contributions/skillForm/SkillForm.component.tsx]: Let's add a documentation for this function

File: src/components/contributions/contributionNav/ContributionNav.component.tsx (hunk 1 above)
Comment (andela-uofoegbu) [inline src/components/contributions/contributionNav/ContributionNav.component.tsx]: @andela-cofor That's the only way to change color of the icon using Paul's react-inline-svg-icon package

Comment (andela-cofor) [inline src/components/contributions/contributionNav/ContributionNav.component.tsx]: Hmmm I see @andela-uofoegbu if that's the only way then, that's fine

Comment (Mbarak-Mbigo) [inline src/components/contributions/contributionNav/ContributionNav.component.tsx]: @andela-cofor it's not styling per se it's a property it's receiving, i.e a `prop`

File: src/components/contributions/skillForm/SkillForm.scss (hunk 2 above)
Comment (Mbarak-Mbigo) [inline src/components/contributions/skillForm/SkillForm.scss]: my bad, I must have forgotten

File: src/components/contributions/skillForm/SkillForm.component.tsx (hunk 3 above)
Comment (Mbarak-Mbigo) [inline src/components/contributions/skillForm/SkillForm.component.tsx]: Cool

File: src/components/contributions/contributionNav/ContributionNav.component.tsx (hunk 1 above)
Comment (andela-cofor) [inline src/components/contributions/contributionNav/ContributionNav.component.tsx]: Noted

**Comment by Mbarak-Mbigo** [comment]
@andela-cofor the number of files that were changed increased when I was requested to make changes that were affecting the ContributionsFormPage. It was about updating the Icons from the ones from Material UI to those provided by Paul's library: `react-inline-svg-icon`

File: src/components/contributions/skillForm/Okbbs.component.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    this.handleOkbbInput = this.handleOkbbInput.bind(this);
+  }
+
+  /**
+   * This is a method that generates a list of okbb items
+   *
+   * @param {Array} okbbList an array of okbb items
+   * @return {any} a list of okbb items
+   */
+
```
Comment (andela-kakpobome) [inline src/components/contributions/skillForm/Okbbs.component.tsx]: ### Critical

Remove this unnecessary space

Comment (Mbarak-Mbigo) [reply src/components/contributions/skillForm/Okbbs.component.tsx]: Done

**Pull request merged**